crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -141,68 +124,45 @@ impl DepthCloudDrawData {
-                crate::profile_scope!("depth");
-
-                let depth_texture_size = wgpu::Extent3d {
-                    width: depth_cloud.depth_dimensions.x,
-                    height: depth_cloud.depth_dimensions.y,
-                    depth_or_array_layers: 1,
-                };
-                let depth_format = match depth_cloud.depth_data {
-                    DepthCloudDepthData::U16(_) => wgpu::TextureFormat::Depth16Unorm,
-                    DepthCloudDepthData::F32(_) => wgpu::TextureFormat::R32Float,
-                };
-                let depth_texture_desc = TextureDesc {
-                    label: "depth_texture".into(),
-                    size: depth_texture_size,
-                    mip_level_count: 1,
-                    sample_count: 1,
-                    dimension: wgpu::TextureDimension::D2,
-                    format: depth_format,
-                    usage: wgpu::TextureUsages::TEXTURE_BINDING | wgpu::TextureUsages::COPY_DST,
-                };
-                let depth_texture = ctx
-                    .gpu_resources
-                    .textures
-                    .alloc(&ctx.device, &depth_texture_desc);
-
-                let format_info = depth_texture_desc.format.describe();
-                let width_blocks =
-                    depth_cloud.depth_dimensions.x / format_info.block_dimensions.0 as u32;
-                let bytes_per_row_unaligned = width_blocks * format_info.block_size as u32;
-
-                {
-                    crate::profile_scope!("write depth texture");
-                    ctx.queue.write_texture(
-                        wgpu::ImageCopyTexture {
-                            texture: &depth_texture.inner.texture,
-                            mip_level: 0,
-                            origin: wgpu::Origin3d::ZERO,
-                            aspect: wgpu::TextureAspect::All,
-                        },
-                        depth_cloud.depth_data.as_bytes(),
-                        wgpu::ImageDataLayout {
-                            offset: 0,
-                            bytes_per_row: Some(
-                                NonZeroU32::new(bytes_per_row_unaligned)
-                                    .expect("invalid bytes per row"),
-                            ),
-                            rows_per_image: None,
-                        },
-                        depth_texture_size,
-                    );
+            let depth_texture = match &depth_cloud.depth_data {
+                // On native, we can use D16 textures without issues.
+                #[cfg(not(target_arch = "wasm32"))]
+                DepthCloudDepthData::U16(data) => {
+                    create_and_upload_texture(ctx, depth_cloud, data.as_slice(), false)
+                }
+                // On the web, OTOH, they are currently broken, see
+                // https://github.com/gfx-rs/wgpu/issues/3537.
+                #[cfg(target_arch = "wasm32")]
```

Comment:
Just to clarify, it's not that they broken it's just that there's no error message? I.e. if that bug is fixed, we still won't have this path working - WebGL does simply not support the Depth16 format 
I.e.  if we ever want to support this we'd need to do something like RG8_UNorm and do the packing in the shader. Not saying we should do that now, but using depth textures are pure sampling texture is kinda exotic after all.